Code/Framework/AzFramework/AzFramework/DocumentPropertyEditor/ExpanderSettings.cpp

```diff
@@ -0,0 +1,201 @@
+        {
+            if (!m_settingsRegistrar.RemoveSettingFromRegistry(m_fullSettingsRegistryPath))
+            {
+                AZ_Warning("DocumentPropertyEditorSettings", false,
+                    "Failed to clean registry state %s before saving", m_fullSettingsRegistryPath.c_str());
+            }
+        }
+
+        if (m_shouldSettingsPersist && !m_expandedElementStates.empty())
+        {
+            SaveExpanderStates();
+        }
+    }
+
+    void ExpanderSettings::SetExpanderStateForRow(const AZ::Dom::Path& rowPath, bool isExpanded)
+    {
+        m_expandedElementStates[GetStringPathForDomPath(rowPath)] = isExpanded;
+    }
+
+    bool ExpanderSettings::GetExpanderStateForRow(const AZ::Dom::Path& rowPath)
+    {
+        auto path = GetStringPathForDomPath(rowPath);
+        if (m_expandedElementStates.contains(path))
+        {
+            return m_expandedElementStates[path];
+        }
+        return false;
+    }
+
+    bool ExpanderSettings::HasSavedExpanderStateForRow(const AZ::Dom::Path& rowPath) const
+    {
+        return m_expandedElementStates.contains(GetStringPathForDomPath(rowPath));
+    }
+
+    void ExpanderSettings::RemoveExpanderStateForRow(const AZ::Dom::Path& rowPath)
+    {
+        m_expandedElementStates.erase(GetStringPathForDomPath(rowPath));
+    }
+
+    void ExpanderSettings::SetCleanExpanderStateCallback(CleanExpanderStateCallback function)
+    {
+        m_cleanExpanderStateCallback = function;
+    }
+
+    LabeledRowDPEExpanderSettings::LabeledRowDPEExpanderSettings(
+        const AZ::DocumentPropertyEditor::DocumentAdapter* adapter,
+        const AZStd::string& settingsRegistryKey,
+        const AZStd::string& propertyEditorName)
+        : ExpanderSettings(adapter, settingsRegistryKey, propertyEditorName)
+    {
+        // Note: labeled row adapters, like the one used in the inspector, can have so many differently named/pathed
+        // elements visible depending on internal state, that it is undesirable to clean them out when switching entities
+        m_cleanExpanderStateCallback = nullptr;
+    }
+
+    ExpanderSettings::PathType LabeledRowDPEExpanderSettings::GetStringPathForDomPath(const AZ::Dom::Path& rowPath) const
+    {
+        AZ_Assert(m_adapter, "must have a valid owning editor to resolve paths!");
+        PathType newPath(AZ::IO::PosixPathSeparator);
+
+        const auto& fullContents = m_adapter->GetContents();
+        AZ::Dom::Path subPath;
+        for (auto pathComponent : rowPath)
+        {
+            subPath.Push(pathComponent);
+            auto rowValue = fullContents[subPath];
+            for (auto arrayIter = rowValue.ArrayBegin(), endIter = rowValue.ArrayEnd(); arrayIter != endIter; ++arrayIter)
+            {
+                auto& currChild = *arrayIter;
+                if (arrayIter->GetNodeName() == AZ::Dpe::GetNodeName<AZ::Dpe::Nodes::Label>())
```

Comment:
currChild child should be used in the if statement over arrayIter if it is going to be assigned here.
```suggestion
                auto& currChild = *arrayIter;
                if (currChild .GetNodeName() == AZ::Dpe::GetNodeName<AZ::Dpe::Nodes::Label>())
```

Reply:
fixed, thanks!